### PR TITLE
[juniper_srx] Improve TCP input

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Add TLS support to TCP input
- Add custom TCP options support

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
cd packages/juniper_srx && elastic-package test
```

## Related issues

- Relates #1878 

## Screenshots

juniper_srx

### CODE REVIEW

File: packages/juniper_srx/data_stream/log/agent/stream/tcp.yml.hbs [hunk 1]
```diff
@@ -9,8 +9,14 @@ tags:
...
+ssl: {{ssl}}
+{{/if}}
 processors:
 {{#if processors}}
 {{processors}}
 {{/if}}
 - add_locale: ~
+{{#if tcp_options}}
+{{tcp_options}}
+{{/if}}
```
Comment (efd6) [inline packages/juniper_srx/data_stream/log/agent/stream/tcp.yml.hbs:22]: nit: newline

**Review by efd6 — COMMENTED** [review]
Will there be a TLS test?

**Comment by taylor-swanson** [comment]
> Will there be a TLS test?

I thought about it originally and was overthinking the complexity of setting that up. I just tested it and it turns out it's trivial to set up and hard code the cert/key pair (we won't test validating certificates). I'll add a TLS test here and to the other packages as well.

**Comment by taylor-swanson** [comment]
TLS test has been added.

**Review by efd6 — APPROVED** [review]

**Pull request merged**